### ISSUE TITLE
Problem with re-ordering property types and deploying them between env.

### ISSUE THREAD

**Issue by HalldorLyngmo** [umbraco/Umbraco.Cloud.Issues#196]
When re-ordering property types between tabs on the Document type there is an issue with them not being extracted correctly on the next env. This leads to a schema mismatch when transferring content.

Reproduction
------------
Two environments that have one document type that looks like this:
![image](https://user-images.githubusercontent.com/42831978/79735749-0aed1a80-82f9-11ea-934c-869ec5831981.png)

Re-order the property types so 1 & 2 switch with 3 & 4. Then Deploy the change to the next env.
![image](https://user-images.githubusercontent.com/42831978/79735911-4e478900-82f9-11ea-967e-2461fbc3eca4.png)

When looking at the document type on the next env. it now looks like this where all properties are in the first tab:
![image](https://user-images.githubusercontent.com/42831978/79736580-4c31fa00-82fa-11ea-97a6-5d60252288d3.png)

The UDA will identical on both env. but there is something going wrong when extracting the property types into the right location:
```
 "PropertyGroups": [
    {
      "Key": "8926ab3e-c2ba-4bd5-baf7-bcaee966beec",
      "Name": "Section 2",
      "SortOrder": 1,
      "PropertyTypes": [
        {
          "Key": "347225d0-863a-4f78-b6dc-d92a743d919b",
          "Alias": "property1",
          "DataType": "umb://data-type/41d4f7f11ba14a1bb57532dda2b3e886",
          "Description": null,
          "Mandatory": false,
          "Name": "Property 1",
          "SortOrder": 0,
          "Validation": null,
          "VariesByCulture": false,
          "VariesBySegment": false
        },
        {
          "Key": "b24ef9bc-e736-42da-b24d-bd2439ad24bd",
          "Alias": "property2",
          "DataType": "umb://data-type/323abf83beaf4510bcb61cc48c285a80",
          "Description": null,
          "Mandatory": false,
          "Name": "Property 2",
          "SortOrder": 1,
          "Validation": null,
          "VariesByCulture": false,
          "VariesBySegment": false
        }
      ]
    },
    {
      "Key": "bb184fcf-ca49-4360-94ed-417a759c2937",
      "Name": "Section 1",
      "SortOrder": 0,
      "PropertyTypes": [
        {
          "Key": "77b51155-e31c-4f60-abed-c333e5498b0d",
          "Alias": "property3",
          "DataType": "umb://data-type/b8fcfa4b1b464b5eb860de4e89f55066",
          "Description": null,
          "Mandatory": false,
          "Name": "Property 3",
          "SortOrder": 1,
          "Validation": null,
          "VariesByCulture": false,
          "VariesBySegment": false
        },
        {
          "Key": "ae96be5f-a353-4db6-807a-f7f957c9d51b",
          "Alias": "property4",
          "DataType": "umb://data-type/1e53a476657f4f2780319b292b1bc5f7",
          "Description": null,
          "Mandatory": false,
          "Name": "Property 4",
          "SortOrder": 2,
          "Validation": null,
          "VariesByCulture": false,
          "VariesBySegment": false
        }
      ]
    }
  ],
```

### Specifics
Versions:

- Umbraco 8.6.0
- Forms 8.4.0
- Deploy 3.4.2

### Expected result
The document types should be identical on both env.

### Actual result
After deployment the document type on the next env. is incorrect. This leads to a Schema mismatch when transferring content.

### Workaround
On the target env. where the document type is incorrect. If you re-order the properties on the document type so it looks identical to the previous env. Then save it. 
This get's the schema in sync and allows the content transfer to progress further.

---
_This item has been added to our backlog [AB#6752](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/6752)_

**Comment by ronaldbarendse** [comment]
Although the issue is not the same as in https://github.com/umbraco/Umbraco.Cloud.Issues/issues/203, the problem seems to be caused by the same thing: not correctly sorting the exported UDA data...

**Comment by hfloyd** [comment]
This seems to be the same issue as I was having with Media properties: https://github.com/umbraco/Umbraco.Cloud.Issues/issues/124

**Comment by clausjensen** [comment]
This will be fixed in the upcoming 3.5.3 release of Deploy 😃
